### DISCUSSION TITLE
When adjusting the network, which parameters to tune?

### DISCUSSION THREAD

**Discussion by EvgenyUgolkov** [PMEAL/OpenPNM#2686] [Q&A]
Dear Team, good day. I found the code which you implemented to test your algorithm on the Berea sandstone CT-scan roughly 2 month ago on this link

https://github.com/PMEAL/OpenPNM/blob/dev/tests/integration/PoreSpyIO_on_Berea.py

The result looks cool. Could you tell me, if, i have another CT-scan with the values of experimental permeability, and my goal is to tune the simulation permeability with the measured permeability, then what parameters should i tune and in what order? From the provided example, i can see that we can manipulate with

```r
'throat.spacing', 'throat.conduit_lengths', and 'throat.hydraulic_size_factors'
```

Is there anything else that we can tune/adjust? In which order should i do it?

Regards and thank you for the help!

Evgeny

**Comment by jgostick** [comment]
Hopefully you don't have to "tune" too much.  PoreSpy returns a large set of geometrical data, and you have some flexibility in how you tell OpenPNM to use it.  The main things would be whether to assume spheres and cylinders or cubes and cuboids, etc, which cascades to how volumes and lengths are calculated, as well as transport conductances.  

How far off of the experimental value are you getting with your extraction?

**Reply by EvgenyUgolkov** [reply]
@jgostick I didn't try anything except Berea yet, just asking for the future, because i will have to do it soon. I wanted to ask regarding your code for Berea. Here you say that "the main thing would be whether to assume spheres and cylinders or cubes and cuboids, etc". Here is the snippet from your code for Berea sandstone
![image](https://user-images.githubusercontent.com/92923525/217452364-9501cddf-7303-4f64-bafc-d97983e0a72e.png)

1. Do i understand correctly that the assignment we are talking about was done on the line

```r
pn.add_model(propname='throat.hydraulic_size_factors',
                 model=op.models.geometry.hydraulic_size_factors.pyramids_and_cuboids)
```

Should we do it through the 'throat.hydraulic_size_factors'? For some reason, before i thought that we should use

```r
pn.add_model_collection(op.models.collections.geometry.spheres_and_cylinders)
```

but here you don't use this command at all. Could you explain, when we use the first option and when we use the second option?

2. Also, why in this code you didn't include "throat.conduit_length"? In which cases do we need to include it?

3. Are there other options of the network with which we can play with and where can i find a list of them?

4. I tried to calculate the volume of pores of the extracted this way network as follows
```r
Vol_void = np.sum(pn['pore.volume'])+np.sum(pn['throat.volume'])
``` 
but got the KeyError: 'throat.volume'. Seems that here we didn't assign it. However, the simulation is still correct. Is it normal? Also, how can i calculate the porosity of the network in this case? I would like to compare it with the porosity of 3D image 

Sorry for the many questions

**Reply by jgostick** [reply]
If you do ``print(pn.models)`` you can see what models are assigned to your network and what properties they are calculating.  It is possible to have 'spheres' and 'cylinders' for volume calcuations and 'pyramids' and 'cuboids' for the size factors if you wish/need to.  The 'models collections' will assign a pre-selected set of models which assumed 'spheres' and 'cyliners' for all properties, as the name suggests.  You can select your individual models of you can use a collection and then overwrite just a few specific ones.  The idea is that through the choice of pore-scale models you can customize the code as needed.

Side note, in V3 we removed the Geometry and Physics classes and replaced them with the ability to put models on the netork and phase directly.  We considered it an upside that it would show users that models are how you customize openpnm, however, I see now that we could use some additional documentation.  I will add it to my todo list to make a notebook about extracting berea, then looking at how the choice of different pore-scale models affects the results.

**Reply by EvgenyUgolkov** [reply]
@jgostick  Many thanks!
